### DISCUSSION TITLE
Add possibility to toggle selection blur behavior

### DISCUSSION THREAD

**Discussion by selrond** [nick-keller/react-datasheet-grid#89] [Feature requests]
Let's say I want to change the value of a specific cell from outside - like formula bar in Google Sheets for instance.

It is possible, but the blur event fires automatically when I click somewhere else outside the datasheet grid, leaving the user with no selected cell while editing its value from the formula bar, which can be confusing.

Is it possible to have an API disabling this behavior?
Something like `enableBlurEventsOnOutsideClick={off}`

Illustration of what I mean:

![datasheet-grid](https://user-images.githubusercontent.com/6603389/144843202-6a3e7122-11c0-4751-85fc-0d92a5e05f7b.png)

- `input` (with the blue border) is active
- cell with red borders should be active while editing the input, but it's not

**Comment by nick-keller** [comment]
Excellent use-case. 
- How do you think it should react when you click outside of everything?
- How should it work with. multiple grid on the same page?
- How should it handle keyboard events?

**Reply by selrond** [reply]
Haven't given it much thought, but—in principle—leaving the `onBlur` implementation to the user should be enough. This way, all of your questions would be up for the consumer to answer _if they want_.

This would require some refactoring, of course.

**Reply by nick-keller** [reply]
I cannot ignore the last point, the grid needs to know what to do when the user presses a key. Let's say the user presses the up arrow, should we move up?
